example/xml/consumer/tests/Service/HttpClientServiceTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace XmlConsumer\Tests\Service;
+
+use Tienvx\PactPhpXml\Model\Options;
+use Tienvx\PactPhpXml\XmlBuilder;
+use XmlConsumer\Service\HttpClientService;
+use PhpPact\Consumer\InteractionBuilder;
+use PhpPact\Consumer\Model\ConsumerRequest;
+use PhpPact\Consumer\Model\ProviderResponse;
+use PhpPact\Standalone\MockService\MockServerConfig;
+use PHPUnit\Framework\TestCase;
+
+class HttpClientServiceTest extends TestCase
+{
+    public function testGetMovies()
+    {
+        $request = new ConsumerRequest();
+        $request
+            ->setMethod('GET')
+            ->setPath('/movies')
+            ->addHeader('Accept', 'text/xml; charset=UTF8');
+
+        $xmlBuilder = new XmlBuilder('1.0', 'UTF-8');
+        $xmlBuilder
+            ->start('movies')
+                ->eachLike('movie')
+                    ->start('title')
+                        ->contentLike('PHP: Behind the Parser')
+                    ->end()
+                    ->start('characters')
+                        ->eachLike('character', [], new Options(examples: 2))
+                            ->start('name')
+                                ->contentLike('Ms. Coder')
+                            ->end()
+                            ->start('actor')
+                                ->contentLike('Onlivia Actora')
+                            ->end()
+                        ->end()
+                    ->end()
+                    ->start('plot')
+                        ->contentLike(
+                            <<<EOF
+                            So, this language. It's like, a programming language. Or is it a
+                            scripting language? All is revealed in this thrilling horror spoof
+                            of a documentary.
+                            EOF
+                        )
+                    ->end()
+                    ->start('great-lines')
+                        ->eachLike('line')
+                            ->contentLike('PHP solves all my web problems')
+                        ->end()
+                    ->end()
+                    ->start('rating', ['type' => 'thumbs'])
```

Comment:
@YOU54F It's your library, what do you think of end-users needing to do this?

Reply:
Do you mean this?

```
->start('rating')
    ->attribute('type', 'thumbs')
    ->contentLike(7)
->end()
```

Reply:
It looks ok to me , and its a decent start. I'm not familiar enough with PHP to know if its feel natural enough for users expected experience, but I can run the tests and get a pact expressing my xml content, with matchers. That capability versus not having it, makes me think whatever works, works for now, and we can iterate in the future.

Working with XML is never going to be nice, especially if you have huge objects, and I think for a better experience overall, if you are using an XML format, such as SOAP, building out a pact-plugin to deal with that might make a better experience.

Do you have reservations with the approach?

As an aside, I know that in Java world, users have difficulty generating pact interaction definitions in the standard library and there are a few DSL builders provided by community, so what is being proposed here isn't alien.

I'm definitely curious if users want slimmed down core package, with the ability to bolt in what they need, when they need so they can mix and match, or is it better just to have everything in one box. I know there will be people on both sides on the fence, and I often flip flop (I might take an OSS project, fork it and strip out extraneous things I don't need, for my purpose - niche case but you know :D)

Reply:
It's totally unnatural in any programming language to need to coordinate start and end. I'd suggest looking at functional options pattern. Basically using a spread operator, with functions which are supplied via spread, receive and modify an internal state object (can be idempotent read-only objects if needs be).

That would mean each function would automatically start and end tags, so it could be implicit, but it's still quite heavy-weight compared to supplying XML, because it's such an open format.

Reply:
Thanks for the suggestion I implemented functional options pattern. Please review again.

This is my first time using that pattern. The code look a bit longer for end-users to write.